mods/dwm-ghost-mods.wh.cpp

```diff
@@ -0,0 +1,499 @@
+// ==WindhawkMod==
+// @id              dwm-ghost-mods
+// @name            DWM Ghost Mods
+// @description     Allows you to use basic or classic theme with your DWM ghost windows and more!
+// @version         1.1
+// @author          ephemeralViolette
+// @github          https://github.com/ephemeralViolette
+// @include         dwm.exe
+// @compilerOptions -ldwmapi -luxtheme
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# DWM Ghost Mods
+
+This mod injects into DWM and allows you to modify the behaviour of DWM ghost (not responding) windows. For
+example, you can restore non-DWM window frames, making this useful for blending in non-responsive windows
+with the basic or classic theme.
+
+![Preview image](https://raw.githubusercontent.com/ephemeralViolette/images/main/dwmghostmods.png)
+
+**This mod should be used with a tool such as BasicThemer or ClassicThemeTray.** It enhances the behaviour of
+these tools, and will not work properly with standard DWM window frames.
+
+Currently, you can:
+- Disable the DWM-rendered window frames on ghost windows.
+- Modify or disable the frost effect introduced with Windows Vista, which lightens the ghost window after
+  sending inputs to it a few times.
+
+# ⚠ Important usage note ⚠
+
+In order to use this mod, you must enable Windhawk to inject into system processes in its advanced settings.
+If you do not do this, it will silently fail to inject. **Changing the Windhawk advanced settings will also
+affect any other mod you have installed, and may cause instability as any other mod that injects into all
+processes will now inject into system processes too.**
+
+This also means that the mod will not work with the portable version of Windhawk.
```

Comment:
Have you tried that? It might work just by running it as administrator.

Reply:
I haven't managed to get it working with the portable version at all. DWM is a protected process that you can only modify using a system account such as `NT AUTHORITY\SYSTEM`, which the standard version runs under since it uses a service.

Furthermore, while I don't suspect many people are manually running the portable version as SYSTEM, I tried that and still couldn't get it working at all.

Reply:
FYI it did just work for me with the portable version by running it as administrator. So your paragraph, which [was also copied to other mods as well](https://github.com/search?q=repo%3Aramensoftware%2Fwindhawk-mods+%22will+not+work+on+portable%22&type=code), doesn't seem to be correct. I only tested with Windhawk v1.4.1, the latest version at this point.

![image](https://github.com/ramensoftware/windhawk-mods/assets/4129781/a7e7bc8c-beaf-4cf2-b9be-52bb24c59e1d)